.github/workflows/testserver.yml

```diff
@@ -0,0 +1,378 @@
+        id: get-labels
+        uses: actions/github-script@v6
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          script: |
+            const response = await github.rest.issues.listLabelsOnIssue({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              issue_number: context.issue.number
+            })
+            const labels = response.data
+            return labels.map(label => label.name)
+
+
+  # Check that the build job has run successfully before deploying
+  check-build-status:
+    needs: [ get-labels ]
+    runs-on: ubuntu-latest
+    # Only run workflow if the added label is a deploy label
+    if: contains(needs.get-labels.outputs.labels, 'deploy:artemis-test')
+    steps:
+      - name: Get latest successful build for branch
+        id: check_build
+        uses: octokit/request-action@v2.x
+        with:
+          route: GET /repos/${{ github.repository }}/actions/workflows/build.yml/runs?event=pull_request&status=success&head_sha=${{ github.event.pull_request.head.sha }}
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+      # Remove deployment-error label if new run is started
+      - uses: actions-ecosystem/action-remove-labels@v1
+        if: fromJSON(steps.check_build.outputs.data).total_count > 0
+        with:
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+          labels: |
+            deployment-error
+
+      # In case of invalid build status, remove deploy labels
+      - uses: actions-ecosystem/action-remove-labels@v1
+        if: fromJSON(steps.check_build.outputs.data).total_count == 0
+        with:
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+          labels: |
+            deploy:artemis-test1
+            deploy:artemis-test2
+            deploy:artemis-test3
+            deploy:artemis-test4
+            deploy:artemis-test5
+            deploy:artemis-test6
+            deploy:artemis-test7
+            deploy:artemis-test8
+            deploy:artemis-test9
+            deploy:artemis-test10
+
+      - name: Check if latest push had successful build
+        if: fromJSON(steps.check_build.outputs.data).total_count == 0
+        uses: actions/github-script@v6
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          script: |
+            github.rest.issues.createComment({
+              issue_number: context.issue.number,
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              body: '### ❌ Unable to deploy to test servers ❌\nThe docker build needs to run through before deploying.'
+            })
+            core.setFailed('The build needs to run through first. Please wait for the build to finish and then try again.')
+
+
+  # Check which test server to deploy to based on the label
+  filter-matrix:
+    needs: [ get-labels, check-build-status ]
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        include:
```

Comment:
Instead of defining all environments in the action you could add small configuration files directly to the repo and load the matrix dynamically. This might ease the future use for new environments.

as reference: https://code.dblock.org/2021/09/03/generating-task-matrix-by-looping-over-repo-files-with-github-actions.html

Reply:
Since I want to get this PR merged soon, so I can create an Iris test server in the develop-iris branch with it, I would push this to a follow-up PR. 
But thanks for the pointer! Would certainly make this file more readable.